components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -403,14 +405,21 @@ func (o *odometry) trackPosition(ctx context.Context) {
 			// Limit the yaw to a range of positive 0 to 360 degrees.
 			o.orientation.Yaw = math.Mod(o.orientation.Yaw, oneTurn)
 			o.orientation.Yaw = math.Mod(o.orientation.Yaw+oneTurn, oneTurn)
-
-			// Calculate X and Y by using centerDist and the current orientation yaw (theta).
-			o.position.X += (centerDist * math.Sin(o.orientation.Yaw))
-			o.position.Y += (centerDist * math.Cos(o.orientation.Yaw))
+			if o.useOri {
+
+				compass := yawToCompassHeading(o.orientation.Yaw)
+				// Calculate X and Y by using centerDist and the current orientation yaw (theta).
+				o.position.X += (centerDist * math.Sin(rdkutils.DegToRad(compass)))
+				o.position.Y += (centerDist * math.Cos(rdkutils.DegToRad(compass)))
+			} else {
+				// Calculate X and Y by using centerDist and the current orientation yaw (theta).
+				o.position.X += -(centerDist * math.Sin(o.orientation.Yaw))
+				o.position.Y += (centerDist * math.Cos(o.orientation.Yaw))
+			}
```

Comment:
```go
angle := o.orientation.Yaw
xFlip := -1.0
if o.useOri {
      angle := rdkutils.DegToRad(yawToCompassHeading(o.orientation.Yaw))
      xFlip := 1.0
}
o.position.X += xFlip*(centerDist * math.Sin(angle))
o.position.Y += (centerDist * math.Cos(angle))
```

I'm assuming if we just flip the calculation for relative x travel we won't need that `xFlip` at all.